### DISCUSSION TITLE
Casting to PHP 8.1 Enums

### DISCUSSION THREAD

**Discussion by squareborg** [spatie/laravel-data#113] [Q&A]
Hey 

Is there built in support for casting to PHP 8.1 enums? I've implemented a custom cast but unsure if thats the correct approach my code

```
class MyEnumCast implements Cast
{
    public function cast(DataProperty $property, mixed $value): mixed
    {
        $enumType = $property->types()->first();
        return $enumType::from($value);
    }
}
class SomeData extends Data
{
    public function __construct(
        #[WithCast(MyEnumCast::class)]
        public SomeEnum|null $the_value,
```

**Comment by rubenvanassche** [comment]
You can use the built in enum cast (https://github.com/spatie/laravel-data/blob/913f74865e82c73a858eea4dc3519b3684a716d2/src/Casts/EnumCast.php#L10). In v2 this cast will not be required anymore.

**Reply by suyar** [reply]
Looking forward to the release of V2.
